### PR TITLE
feature(digital-products): Users should be able to buy digital products:

### PR DESCRIPTION
**Pull request by codestaintin**
#### What does this PR do?

- Enable buyers and sellers to be able to buy/sell digital Products

 #### Description of Task to be completed?

- add product type to schema
- add product category to schema
- add option of choosing digital product on the front-end
- add download link after buyer checkout for digital products
- Fix payment gateway bug
- Add cloudinary presets

 #### How should this be manually tested?

- Clone the repo - `$ git clone https://github.com/andela/k2-rc.git`
- Run ```git checkout feature/153812328/digital-product```
- Follow the project README to set up the application on your machine
- As as admin or shop owner, visit the admin dashboard and add a digital product by choosing product type and product category
- As a buyer purchase a digital product and checkout, a download link is visible after payment is successful

#### What are the relevant pivotal tracker stories?

- story type: feature
- story id: 153812328
- story title: Digital Products

#### Screenshots (if appropriate)
screen shot 2018-01-30 at 7 47 40 am
screen shot 2018-01-30 at 1 39 20 pm

#### Questions

N/A

[Finished #153812328]

### CODE REVIEW

File: server/methods/core/cart.js [hunk 1]
```diff
@@ -373,12 +377,12 @@ Meteor.methods({
     // const product = Collections.Products.findOne(productId);
```
Comment (damakahindi) [inline server/methods/core/cart.js]: Remove commented out code.

File: server/methods/core/cart.js [hunk 2]
```diff
@@ -840,7 +847,7 @@ Meteor.methods({
       }
     }
     if (!updated) {
-      // if we didn't do one of the inline updates, then run the update here
+      // if we didn"t do one of the inline updates, then run the update here
```
Comment (damakahindi) [inline server/methods/core/cart.js]: Why were these replaced?

Comment (codestaintin) [reply server/methods/core/cart.js]: I will fix and revert that now. Thanks for pointing that out

File: server/methods/core/cart.js (hunk 1 above)
Comment (codestaintin) [reply server/methods/core/cart.js]: Alright

**Pull request merged**